File: 2625-increment-submatrices-by-one/increment-submatrices-by-one.cs
public class Solution {
    public int[][] RangeAddQueries(int n, int[][] queries) {
        var mat = new int[n][];

        for (int i = 0; i < n; i++) {
            mat[i] = new int[n];
        }

        for (int i = 0; i < queries.Length; i++) {
            int row1 = queries[i][0], row2 = queries[i][2];
            int col1 = queries[i][1], col2 = queries[i][3];

            for (int j = row1; j <= row2; j++) {
                mat[j][col1]++;
                if (col2 + 1 < n) {
                    mat[j][col2 + 1]--;
                }
            }
        }

        for (int i = 0; i < n; i++) {
            for (int j = 1; j < n; j++) {
                mat[i][j] += mat[i][j - 1];
            }
        }

        return mat;
    }
}

File: 2625-increment-submatrices-by-one/increment-submatrices-by-one.py
import numpy as npy

class Solution:
    def rangeAddQueries(self, n: int, queries: List[List[int]]) -> List[List[int]]:
        temp = [[0] * n for _ in range(n)]
        mat = npy.array(temp)

        for x in queries:
            row1, row2 = x[0], x[2]
            coloumn1, coloumn2 = x[1], x[3]

            mat[row1 : row2 + 1, coloumn1 : coloumn2 + 1] += 1

        return mat.tolist()
